File: learn-php/ISBN.php
<?php

/* berlinix, 2012/07/25
 * ISBN checksum algo
 *
 * ISBN EAN*UCC-组区号-出版者号-出版序号-效验码
 */

/* length of isbn: 13, here is 12 */
#$isbn = "978711526431";
$isbn = "978711526431";
$varr = "131313131313";

$len = strlen($isbn);
$sum = 0;
for($i = 0; $i < $len; ++$i) {
    $sum += intval($isbn[$i]) * intval($varr[$i]);
}

$sret = 10 - $sum % 10;

echo "$sret\n"

?>

File: learn-cpp/ISBN.cpp
#include <iostream>
#include <cstring>

int isbn_cs(const char* isbn)
{
    const char* varr = "131313131313";
    const int len = 12;

    int sum = 0;
    for(int i = 0; i < len; ++i) {
        sum += (int(isbn[i] - '0') * int(varr[i] - '0'));
    }

    int checksum = 10 - sum % 10;

}

int main(int argc, const char** argv)
{
    if(argc != 2) {
        printf("usage: ./a.out isbn");
        return 0;
    }

    const char* isbn = argv[1]; 
    int checksum = isbn_cs(isbn);
    printf("%s-%d\n", isbn, checksum);

    return 0;
}
